src/main/scala/org/broadinstitute/dsde/rawls/expressions/SlickExpressionParsing.scala

```diff
@@ -207,17 +210,17 @@ trait SlickExpressionParser extends JavaTokenParsers {
       (entity, ordering) <- queryPipeline.get
       entityAttribute <- entityAttributeQuery if entity.id === entityAttribute.entityId
       attribute <- attributeQuery if entityAttribute.attributeId === attribute.id && attribute.name === attributeName
-    } yield (attribute, ordering)).result
+    } yield (entity.id, attribute, ordering)).result
 
 
     attributeForNameQuery.map { entityWithAttributeRecs =>
```

Comment:
This entire code block 216-223 is mysterious to me.  Please comment extensively on what this does.